images/plugin-compiler/README.md

```diff
@@ -0,0 +1,33 @@
+# Tyk Plugin compiler
+
+## Building a plugin
+Navigate to where your plugin is and build using a docker volume to
+mount your code into the image. Since the vendor directory needs to be
+identical between the gateway build and the plugin build, this means
+that you should pull the version of this image corresponding to the
+gateway version you are using.
+
+This also implies that if your plugin has vendored modules that are
+also used by Tyk gateway then your module will be overridden by the
+version that Tyk uses.
+
+```shellsession
+% docker run -v `pwd`:/plugin-source tykio/tyk-plugin-compiler:v2.9.4.2 myplugin.so
+```
+
+You will find a myplugin.so in the current directory which is the file
+that goes into the API definition
+
+## Plugin aliasing
+Plugins are loaded via `dl_open(3)` and the shared library cache,
+`ld.so.cache` will be used. Therefore,even if a plugin's content
+changes but the filename does not, the cached plugin will be used.
+
+See the manpages for `dl_open(3)` and `ld.so(8)` on your platform for
+more details.
+
+## Hot reloading
+Using `/tyk/reload/group` will _not_ update the plugin if you have
+compiled a fresh version and used the same pathname.
+
+To reload a plugin, you will have to restart the gateway process.
```

Comment:
This could open a sea of complaints if that’s the only way @alephnull

Reply:
This is hard to fix unless we implement our own dynamic loading logic. We should wait for the `plugin` package to implement something that allows the safe unloading/reloading of Go shared libraries, etc.
For now the only strategy is to use different filenames every time the code changes or perform a full process reload.